autoharness/agent.py

```
@@ -0,0 +1,155 @@
+- Verify your solution before finishing.
+"""
+
+MODEL = "gpt-5"
+MAX_TURNS = 30
+
+
+@function_tool
+def run_shell(command: str) -> str:
+    """Execute a shell command and return combined stdout+stderr."""
+    try:
+        result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)